database/cassandra/cassandra.go

```diff
@@ -120,17 +120,25 @@ func (c *Cassandra) Open(url string) (database.Driver, error) {
 		cluster.Timeout = timeout
 	}
 
-	if len(u.Query().Get("sslmode")) > 0 && len(u.Query().Get("sslrootcert")) > 0 && len(u.Query().Get("sslcert")) > 0 && len(u.Query().Get("sslkey")) > 0 {
-		if u.Query().Get("sslmode") != "disable" {
-			cluster.SslOpts = &gocql.SslOptions{
-				CaPath:   u.Query().Get("sslrootcert"),
-				CertPath: u.Query().Get("sslcert"),
-				KeyPath:  u.Query().Get("sslkey"),
-			}
-			if u.Query().Get("sslmode") == "verify-full" {
-				cluster.SslOpts.EnableHostVerification = true
-			}
+	if len(u.Query().Get("sslmode")) > 0 && u.Query().Get("sslmode") != "disable" {
+		var sslOpts gocql.SslOptions
+
+		switch u.Query().Get("sslmode") {
+		case "noverify":
+			// Do nothing, defaults are fine
+		case "verify-ca":
+			sslOpts.CaPath = u.Query().Get("sslrootcert")
+		case "verify-full":
+			sslOpts.CaPath = u.Query().Get("sslrootcert")
+			sslOpts.CertPath = u.Query().Get("sslcert")
```

Comment:
`CertPath` and `KeyPath` configure the client cert/key path thus they should be settable independently of the server side cert verification option.

Reply:
I'm no longer using this. If you want these changes merged then I'll leave that to you. If not, feel free to close.